lychee-bin/src/parse.rs

```diff
@@ -13,8 +13,20 @@ pub(crate) fn parse_remaps(remaps: &[String]) -> Result<Remaps> {
         .context("Remaps must be of the form '<pattern> <uri>' (separated by whitespace)")
 }
 
-pub(crate) fn parse_base(src: &str) -> Result<Base, lychee_lib::ErrorKind> {
-    Base::try_from(src)
+pub(crate) fn parse_base(src: &str) -> Result<Base> {
+    match Base::try_from(src) {
+        Ok(x) => Ok(x),
+        Err(e) => {
+            // if context is defined, clap displays only the context string in
+            // argument parse errors. to keep the message from within InvalidBase,
+            // we need to retain it manually.
+            let message = format!(
+                "{e}. See `--help` for more information. If you want to resolve \
+                root-relative links in local files, also see `--root-dir`."
+            );
+            Err(e).context(message)
```

Comment:
So you mean this doesn't work?

```rust
pub(crate) fn parse_base(src: &str) -> Result<Base> {
    Base::try_from(src).context(
        "See `--help` for more information. If you want to resolve \
        root-relative links in local files, also see `--root-dir`."
    )
}
```

Reply:
Yeah, that would only print the context message without anything from the original error. it looks like this:
```
error: invalid value 'a:datafdajsio' for '--base-url <BASE_URL>': See `--help` for more information. If you want to resolve root-relative links in local files, also see `--root-dir`.
```